test/tools.js

```diff
@@ -35,174 +35,6 @@ const findFreePort = () => {
         });
 };
 
-const testUrl = (url, expected) => {
```

Comment:
These unit tests covered plenty of situations (e.g. IPv6 and other special URLs) that are not covered by the main unit tests. Can we add those tests?

Reply:
This PR is using WHATWG URLs directly. Can you elaborate please?